buildSrc/src/main/java/gradlebuild/NcursesPlugin.java

```diff
@@ -30,21 +36,26 @@
         addPlatform(platformContainer, "linux_aarch64_ncurses6", "linux", "aarch64");
     }
 
-    @Mutate void configureBinaries(@Each NativeBinarySpecInternal binarySpec, NcursesVersion ncursesVersion) {
+    @Mutate void configureBinaries(@Each NativeBinarySpecInternal binarySpec, List<NcursesVersion> ncursesVersions) {
         NativePlatform targetPlatform = binarySpec.getTargetPlatform();
         if (targetPlatform.getOperatingSystem().isLinux() && targetPlatform.getName().contains("ncurses")) {
-            if (!targetPlatform.getName().contains("ncurses" + ncursesVersion.getNcursesVersion())) {
+            if (!ncursesVersions.stream().anyMatch(ncursesVersion -> isNcursesVersion(targetPlatform, ncursesVersion.getNcursesVersion()))) {
                 binarySpec.setBuildable(false);
             }
         }
         if (binarySpec.getComponent().getName().contains("Curses")) {
-            if (targetPlatform.getOperatingSystem().isLinux() && !ncursesVersion.getNcursesVersion().equals("5")) {
+            if (targetPlatform.getOperatingSystem().isLinux() && !isNcursesVersion(targetPlatform, "5")) {
                 binarySpec.getLinker().args("-lncursesw");
             } else {
                 binarySpec.getLinker().args("-lcurses");
             }
         }
     }
+
+    private boolean isNcursesVersion(NativePlatform targetPlatform, String ncursesVersion) {
```

Comment:
Nit: let's pass the `NcursesVersion` object here. It would also be nicer to rename `NcursesVersion.getNcursesVersion()` to something like `getVersionNumber()` or something less redundant.